packages/mdl-menu/simple/foundation.js

```diff
@@ -0,0 +1,201 @@
+        this.adapter_.setTransitionDelay(item, `${itemDelay}s`);
+      }
+    }
+  }
+
+  // Remove transition delays from menu items.
+  removeTransitionDelays_() {
+    const items = this.adapter_.getItems();
+    if (items) {
+      for (let i = 0; i < items.length; i++) {
+        this.adapter_.setTransitionDelay(items[i], null);
+      }
+    }
+  }
+
+  // Animate menu opening or closing.
+  animationLoop_() {
+    const time = window.performance.now();
+    const {TRANSITION_DURATION_MS, TRANSITION_X1, TRANSITION_Y1, TRANSITION_X2, TRANSITION_Y2,
+        TRANSITION_SCALE_ADJUSTMENT_X, TRANSITION_SCALE_ADJUSTMENT_Y} = MDLSimpleMenuFoundation.numbers;
+    const currentTime = clamp((time - this.startTime_) / TRANSITION_DURATION_MS);
+
+    let currentTimeX;
+    let currentTimeY;
+
+    let startScaleY = this.startScaleY_;
+    if (this.targetScale_ === 1) {
+      // Start with the menu at the height of a single item.
+      startScaleY = this.itemHeight_ ? Math.max(this.itemHeight_ / this.dimensions_.height, startScaleY) : startScaleY;
+      // X axis moves faster, so time-shift forward.
+      currentTimeX = clamp(currentTime + TRANSITION_SCALE_ADJUSTMENT_X);
+      // Y axis moves slower, so time-shift backwards and adjust speed by the difference.
+      currentTimeY = clamp((currentTime - TRANSITION_SCALE_ADJUSTMENT_Y) / (1 - TRANSITION_SCALE_ADJUSTMENT_Y));
+    } else {
+      // Animate X axis very slowly, so that only the Y axis animation is visible during fade-out.
+      currentTimeX = clamp((currentTime - TRANSITION_SCALE_ADJUSTMENT_X) / (1 - TRANSITION_SCALE_ADJUSTMENT_X));
+      // No time-shifting on the Y axis when closing.
+      currentTimeY = currentTime;
+    }
+
+    // Apply cubic bezier easing independently to each axis.
+    const easeX = bezierProgress(currentTimeX, TRANSITION_X1, TRANSITION_Y1, TRANSITION_X2, TRANSITION_Y2);
+    const easeY = bezierProgress(currentTimeY, TRANSITION_X1, TRANSITION_Y1, TRANSITION_X2, TRANSITION_Y2);
+
+    // Calculate the scales to apply to the outer container and inner container.
+    this.scaleX_ = this.startScaleX_ + (this.targetScale_ - this.startScaleX_) * easeX;
+    const invScaleX = 1 / (this.scaleX_ === 0 ? 1 : this.scaleX_);
+    this.scaleY_ = startScaleY + (this.targetScale_ - startScaleY) * easeY;
+    const invScaleY = 1 / (this.scaleY_ === 0 ? 1 : this.scaleY_);
+
+    // Apply scales.
+    this.adapter_.setScale(this.scaleX_, this.scaleY_);
+    this.adapter_.setInnerScale(invScaleX, invScaleY);
+
+    // Stop animation when we've covered the entire 0 - 1 range of time.
+    if (currentTime < 1) {
+      requestAnimationFrame(() => this.animationLoop_());
+    } else {
+      this.running_ = false;
+      this.adapter_.removeClass(MDLSimpleMenuFoundation.cssClasses.ANIMATING);
+    }
+  }
+
+  // Starts the open or close animation.
+  animateMenu_() {
+    this.startTime_ = window.performance.now();
```

Comment:
_ultra_ nit: You _may_ want to consider moving this within the initial rAF for super precise timing, and also if for some reason this code runs but the rAF loop is deferred (e.g. user is on another tab, etc). It probably doesn't matter though.